sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -441,7 +441,15 @@ class Analyzer(
   object ResolveRelations extends Rule[LogicalPlan] {
     private def lookupTableFromCatalog(u: UnresolvedRelation): LogicalPlan = {
       try {
-        catalog.lookupRelation(u.tableIdentifier, u.alias)
+        val table = catalog.lookupRelation(u.tableIdentifier, u.alias)
+        table match {
+          case SubqueryAlias(_, s: SimpleCatalogRelation)
```

Comment:
Will follow your suggestions. So far, we have a bug in `resolved` of `InsertIntoTable`. Thus, `CheckAnalysis` does not work. Let me fix it.